xc/device.ml

```diff
@@ -2787,11 +2787,22 @@ module Backend = struct
                       (string_of_message msg)
 
     let qmp_event_thread () =
-      debug "Starting QMP_Event thread";
-      (* Add the existing qmp sockets first *)
-      Sys.readdir var_run_xen_path
-      |> Array.to_list
-      |> List.iter (fun x -> try Scanf.sscanf x "qmp-event-%d" add with _ -> ());
+      let qmp_domid socket =
+        try Some (Scanf.sscanf socket "qmp-event-%d" (fun d -> d))
+        with _ -> None in
+
+      begin
+        try
+          debug "Starting QMP_Event thread";
+          (* Add the existing qmp sockets first *)
+          Sys.readdir var_run_xen_path
+          |> Array.to_list
+          |> List.filter_map qmp_domid
+          |> List.iter add
```

Comment:
What if a few adds fail, but the majority would succeed? Should we do a best effort here and connect to as many as we can and then log and ignore the errors for the rest? (add does log the exception but then rethrows)

Reply:
(and then if we add a timeout inside `add` in the future, this would do the appropriate thing there as well)

Reply:
So the idea is to catch exceptions escaping from `add`, log them, but continue. I like that.